src/wallet/coin_selection.rs

```diff
@@ -57,9 +57,8 @@
 //!                 },
 //!             )
 //!             .collect::<Vec<_>>();
-//!         let additional_fees = additional_weight as f32 * fee_rate.as_sat_vb() / 4.0;
-//!         let amount_needed_with_fees =
-//!             (fee_amount + additional_fees).ceil() as u64 + amount_needed;
+//!         let additional_fees = wallet::fee_wu(fee_rate, additional_weight);
```

Comment:
Sorry, I only noticed this after approving the changes. Shouldn't this be changed as well to use the new methods?

Reply:
Oh boy, seems I still don't know how to run cargo in a way that catches everything. What's your workflow, what cargo command do you run to check your work?

Reply:
I've squashed these two commits into a single one since the first patch was really a step in the development process (after implementing your suggestions @afilini). Since it may not be fresh in mind when merging, and/or others are reviewing, I think it is less review work to have a single patch.

Reply:
Apart from fixing the document code call site, no other changes were made while squashing. The patch is reasonably short now so its not too hard to verify the patch is ok again. Please re-review. Also, please say if this squashing is not what you like.

Reply:
Please review carefully the two changes that look like:
```
-        fee_amount += calc_fee_bytes(serialize(&drain_output).len() * 4);
+        fee_amount += fee_rate.fee_vb(serialize(&drain_output).len());
```
Thanks.

Reply:
> What's your workflow, what cargo command do you run to check your work?

I'm not sure what went wrong here, usually just running `cargo test` or `cargo test --doc` should have caught it. Weird.